pkg/controller/atlascustomrole/custom_role.go

```diff
@@ -0,0 +1,106 @@
+package atlascustomrole
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation/customroles"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/workflow"
+)
+
+type roleController struct {
+	ctx       *workflow.Context
+	projectID string
+	service   customroles.CustomRoleService
+	role      *akov2.AtlasCustomRole
+	dpEnabled bool
+}
+
+func handleCustomRole(ctx *workflow.Context, akoCustomRole *akov2.AtlasCustomRole, dpEnabled bool) workflow.Result {
+	ctx.Log.Debug("starting custom role processing")
+	defer ctx.Log.Debug("finished custom role processing")
+
+	r := roleController{
+		ctx:       ctx,
+		service:   customroles.NewCustomRoles(ctx.SdkClient.CustomDatabaseRolesApi),
+		role:      akoCustomRole,
+		dpEnabled: dpEnabled,
+	}
+
+	return r.Reconcile()
+}
```

Comment:
I am not sure why we need sub-reconciler for the custom role.

This seems to me like the regular inner reconcile logic we always have, after usual login that gets the Kubernetes object from the plain request, skips if needed, gets the Atlas status, etc. Not sure why we have to separate it here.

But even if we separate, we do not need a custom `roleController` for something that is request specific. Do we?

This separation makes more sense when you have a list of items at the top level controller, and then at the next level here there is the item to be reconciled.